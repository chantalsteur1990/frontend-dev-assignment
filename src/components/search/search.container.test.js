import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import SearchContainer from './search.container';

configure({ adapter: new Adapter() });

const mockProps = {
  name: 'search',
  placeholder: 'Zoeken',
  ariaLabelInput: 'zoeken',
  ariaLabelClear: 'annuleren',
  ariaLabelSubmit: 'zoeken',
  handleSearch: jest.fn(),
  handleSuggestions: jest.fn(),
  handleQueryChange: jest.fn(),
  value: ''
};

describe('Search container', () => {
  it('should render (snapshot)', () => {
    const component = shallow(<SearchContainer {...mockProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it ('should call the submit callback', () => {
    const component = shallow(<SearchContainer {...mockProps} />);
    component.instance().handleSubmit();
    expect(mockProps.handleSearch).toHaveBeenCalled();
  })
});
