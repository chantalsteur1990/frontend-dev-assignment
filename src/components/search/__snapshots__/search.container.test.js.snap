// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search container should render (snapshot) 1`] = `
<Search
  ariaLabelClear="annuleren"
  ariaLabelInput="zoeken"
  ariaLabelSubmit="zoeken"
  handleChange={[Function]}
  handleClear={[Function]}
  handleKeyUp={[Function]}
  handleQueryChange={[Function]}
  handleSearch={[Function]}
  handleSubmit={[Function]}
  handleSuggestions={[Function]}
  name="search"
  placeholder="Zoeken"
  value=""
/>
`;
